src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,86 @@
 import React from 'react';
 import './CurrentUser.scss';
+import { getUser } from '../../api/api';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  userId: number,
+  onClear: () => void,
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+type State = {
+  user: User | null,
+  userError: boolean,
+};
+
+export class CurrentUser extends React.Component<Props, State> {
+  state = {
+    user: {} as User,
+    userError: false,
+  };
+
+  async componentDidMount() {
+    this.loadUser();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.userId !== prevProps.userId) {
+      this.loadUser();
+    }
+  }
+
+  loadUser = async () => {
+    try {
+      const user = await getUser(this.props.userId);
+
+      this.setState({
+        user,
+        userError: false,
+      });
+    } catch {
+      this.setState({
+        userError: true,
+      });
+    }
+  };
+
+  render() {
+    return (
+      !this.state.userError ? (
+        <div className="CurrentUser">
+          <h2 className="CurrentUser__title">
+            <span>
+              Selected user:
+              {' '}
+              {this.state.user.id}
+            </span>
+          </h2>
+
+          <h3 className="CurrentUser__name">
+            {this.state.user.name}
+          </h3>
+
+          <p className="CurrentUser__email">
+            {this.state.user.email}
+          </p>
+
+          <p className="CurrentUser__phone">
+            {this.state.user.phone}
+          </p>
+
+          <button
+            type="button"
+            className="button"
+            onClick={() => {
+              this.props.onClear();
+            }}
+          >
+            Clear user
+          </button>
+        </div>
+      )
+        : (
+          `User id #${this.state.user.id} not found`
+        )
```

Comment:
Чтобы исправить баг №1, нужно здесь использовать не поле id из объекта user (такого поля может и не быть), а поле userId из Props, которые ты сюда передаешь. То есть, соответствующий id юзера из массива todos.

```suggestion
        : (
          `User id #${this.props.userId} not found`
        )
```